handler.go

```diff
@@ -159,17 +154,45 @@ func (p proxyHandler) handleConnectRequest(ctx *Context, rw http.ResponseWriter,
 	}
 	if session.Hijacked() {
 		log.Debugf("martian: connection hijacked by response modifier")
-		return nil
+		return
 	}
 
 	if res.StatusCode != 200 {
 		if cerr == nil {
 			log.Errorf("martian: CONNECT rejected with status code: %d", res.StatusCode)
 		}
 		writeResponse(rw, res)
-		return nil
+		return
+	}
+
+	if req.ProtoMajor == 1 {
+		res.ContentLength = -1
 	}
 
+	if err := p.tunnel("CONNECT", rw, req, res, cw, cr); err != nil {
+		log.Errorf("martian: CONNECT tunnel: %v", err)
+		panic(http.ErrAbortHandler)
```

Comment:
I'm not so sure about the panics. Correct me if I'm wrong, this looks like the proxy would panic if it can't write to the connection.

```go
switch req.ProtoMajor {
	case 1:
                ...
		if err := res.Write(brw); err != nil {
			return fmt.Errorf("got error while writing response back to client: %w", err)
		}
		if err := brw.Flush(); err != nil {
			return fmt.Errorf("got error while flushing response back to client: %w", err)
		}
		if err := drainBuffer(cw, brw.Reader); err != nil {
			return fmt.Errorf("got error while draining buffer: %w", err)
		}
                ...
```

Reply:
This is a special throw catch mechanism in http.Server

```
ErrAbortHandler is a sentinel panic value to abort a handler. While any panic from ServeHTTP aborts the response to the client, panicking with ErrAbortHandler also suppresses logging of a stack trace to the server's error log.
```